drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -2693,6 +2693,10 @@ void RasterizerSceneGLES3::_render_list_template(RenderListParameters *p_params,
 			}
 			glBindTexture(GL_TEXTURE_CUBE_MAP, texture_to_bind);
 		}
+
+		if (get_debug_draw_mode() == RS::VIEWPORT_DEBUG_DRAW_UNSHADED) {
+			shader_variant = SceneShaderGLES3::MODE_UNSHADED;
+		}
```

Comment:
We shouldn't be using shader variants for this. We should just do what we do in the RD renderers and avoid passing in any lighting information. We already avoid lights and reflectionprobes. So all that is left is making sure there is no ambient lighting from the sky:
https://github.com/godotengine/godot/blob/89cc635c0554cb2e518c830969ca4c5eedda0f4e/servers/rendering/renderer_rd/storage_rd/render_scene_data_rd.cpp#L116-L124

We need to avoid shader variants as they are always compiled. The change you made in scene.glsl would increase shader compilation time by 50% for all scene shaders (even if this debug draw mode is never used)